x-pack/plugins/ml/server/plugin.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, IScopedClusterClient, Logger, PluginInitializerContext } from 'src/core/server';
+import { LicenseCheckResult, PLUGIN_ID, PluginsSetup } from './types';
+
+// @ts-ignore: could not find declaration file for module
+import { elasticsearchJsPlugin } from './client/elasticsearch_ml';
+import { makeMlUsageCollector } from './lib/ml_telemetry';
+import { initMlServerLog } from './client/log';
+import { addLinksToSampleDatasets } from './lib/sample_data_sets';
+
+import { annotationRoutes } from './routes/annotations';
+import { calendars } from './routes/calendars';
+import { dataFeedRoutes } from './routes/datafeeds';
+import { dataFrameAnalyticsRoutes } from './routes/data_frame_analytics';
+import { dataRecognizer } from './routes/modules';
+import { dataVisualizerRoutes } from './routes/data_visualizer';
+import { fieldsService } from './routes/fields_service';
+import { fileDataVisualizerRoutes } from './routes/file_data_visualizer';
+import { filtersRoutes } from './routes/filters';
+import { indicesRoutes } from './routes/indices';
+import { jobAuditMessagesRoutes } from './routes/job_audit_messages';
+import { jobRoutes } from './routes/anomaly_detectors';
+import { jobServiceRoutes } from './routes/job_service';
+import { jobValidationRoutes } from './routes/job_validation';
+import { notificationRoutes } from './routes/notification_settings';
+import { resultsServiceRoutes } from './routes/results_service';
+import { systemRoutes } from './routes/system';
+
+declare module 'kibana/server' {
+  interface RequestHandlerContext {
+    ml?: {
+      mlClient: IScopedClusterClient;
+    };
+  }
+}
+
+export class MlServerPlugin {
+  private readonly pluginId: string = PLUGIN_ID;
+  private log: Logger;
+  private version: string;
+
+  private licenseCheckResults: LicenseCheckResult = {
+    isAvailable: false,
+    isActive: false,
+    isEnabled: false,
+    isSecurityDisabled: false,
+  };
+
+  constructor(ctx: PluginInitializerContext) {
+    this.log = ctx.logger.get();
+    this.version = ctx.env.packageInfo.branch;
+  }
+
+  public setup(coreSetup: CoreSetup, plugins: PluginsSetup) {
+    let sampleLinksInitialized = false;
+
+    plugins.features.registerFeature({
+      id: 'ml',
```

Comment:
all 'ml' strings in this file could be `PLUGIN_ID` from `types.ts`

Reply:
Good call - added in 609be5c4c218d1feed4b249158c438ce7c30f99d